src/main/java/piglatin/PigLatinConverter.java

```diff
@@ -38,6 +37,25 @@ public static String convertWordToPigLatin(String word) {
 		return output;            
 	}
 	
+	public static List<String> convertWordFromPigLatin(String word) {		
+        List<String> outputs = new ArrayList<String>(); 
+		if (word.substring(word.length()-3, word.length()).equals("way")){			
+			outputs.add(word.substring(0, word.length()-3));
+			outputs.add("w" + word.substring(0, word.length()-3));
+		} else if (word.substring(word.length()-2, word.length()).equals("ay")){			
+			int lastConsonantIndex = word.length()-3;
+			while(!isVowel(word.charAt(lastConsonantIndex))) {				
+				String candidateWord = word.substring(lastConsonantIndex, word.length()-2)+word.substring(0,lastConsonantIndex);				
+				outputs.add(candidateWord);
+				lastConsonantIndex--;
+			}			
+		} else {
+			throw new IllegalArgumentException(word + " is not a valid pig latin word!");
```

Comment:
It will crash! But given that we in the other cases only write to system.out, I have changed to this here also....